segmenters/nlp/PDFPlumberSegmenter/__init__.py

```diff
@@ -0,0 +1,97 @@
+import io
+from typing import Dict, List
+
+import fitz
+import numpy as np
+import pdfplumber
+from jina.executors.decorators import single
+from jina.executors.segmenters import BaseSegmenter
+from pdftitle import get_title_from_file, get_title_from_io
+
+
+class PDFPlumberSegmenter(BaseSegmenter):
+    """
+    :class:`PDFPlumberSegmenter` Extracts data (text and images) from PDF files.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    @single(slice_nargs=3)
+    def segment(self, uri: str, buffer: bytes, mime_type: str, *args, **kwargs) -> List[Dict]:
```

Comment:
Too much is happening in this function. Please refactor the text extraction and the image extraction in separate functions.

Reply:
refactored, thx!